src/lib/hash.ts

```diff
@@ -192,3 +194,40 @@ class TokenSymbol extends Struct(TokenSymbolPure) {
 function emptyReceiptChainHash() {
   return emptyHashWithPrefix('CodaReceiptEmpty');
 }
+
+function buildSHA(length: 224 | 256 | 384 | 512, nist: boolean) {
+  return {
+    hash(message: (Field | UInt8)[]) {
```

Comment:
I think we should only allow `UInt8` as arguments to make it clear SHA only works with bytes